pkg/minishift/registration/redhat.go

```diff
@@ -0,0 +1,110 @@
+/*
+Copyright (C) 2016 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package registration
+
+import (
+	"strings"
+	"bufio"
+	"fmt"
+	"os"
+	"syscall"
+	"golang.org/x/crypto/ssh/terminal"
+	"github.com/docker/machine/libmachine/drivers"
+)
+
+
+func init() {
+	Register("Redhat", &RegisteredRegistrator{
+		New: NewRedhatRegistrator,
+	})
+}
+
+func NewRedhatRegistrator(d drivers.Driver) Registrator {
+	return &RedhatRegistrator{
+		SSHCommander:   GenericSSHCommander{Driver: d},
+		Driver: d,
+	}
+}
+
+type RedhatRegistrator struct {
+	SSHCommander
+	Driver drivers.Driver
+}
+
+type GenericSSHCommander struct {
+	Driver drivers.Driver
+}
+
+func (sshCmder GenericSSHCommander) SSHCommand(args string) (string, error) {
+	return drivers.RunSSHCommandFromDriver(sshCmder.Driver, args)
+}
+
+func credentials() (string, string) {
```

Comment:
This function will not be part of final PR, currently it's there to test the registration works as expected and we have plan to pass username/password from the command line